Sort/Program.cs

```diff
@@ -0,0 +1,173 @@
+        /// </summary>
+        /// <param name="lhs">左辺値</param>
+        /// <param name="rhs">左辺値</param>
+        /// <returns>true or false</returns>
+        public static bool operator> (Number lhs, Number rhs)
+        {
+            return lhs.Value > rhs.Value;
+        }
+    }
+
+    /// <summary>
+    /// 数列を作るクラス
+    /// </summary>
+    public class Numbers
+    {
+        /// <summary>
+        /// 数字を保持しておく変数
+        /// </summary>
+        List<Number> numbers;
+
+        /// <summary>
+        /// インスタンス生成時にListを初期化します
+        /// </summary>
+        public Numbers()
+        {
+            numbers = new List<Number>();
+        }
+
+        /// <summary>
+        /// Listに数字を追加します
+        /// </summary>
+        /// <param name="number">追加する数字(int)</param>
+        public void Append(int? number)
+        {
+            numbers.Add(number);
+        }
+
+        /// <summary>
+        /// Listに数字を追加します
+        /// </summary>
+        /// <param name="number">追加する数字(Number)</param>
+        public void Append(Number number)
+        {
+            numbers.Add(number);
+        }
+
+        /// <summary>
+        /// 数列を全てアウトプットします
+        /// </summary>
+        public void Output()
+        {
+            numbers.ForEach
+                (
+                num => Console.WriteLine(num.Value)
+                );
+        }
+
+        /// <summary>
+        /// 数列の数字をソートします
+        /// </summary>
+        public void SortValue()
+        {
+            for (int i = 0; i < numbers.Count - 1; i++)
+            {
+                for (int j = numbers.Count - 1; j >= i + 1; j--)
+                {                
+                    if (numbers[j] < numbers[j - 1])
+                    { 
+                        int? temp = numbers[j].Value;
+                        numbers[j] = new Number(numbers[j - 1].Value);
+                        numbers[j - 1] = new Number(temp);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// 引数で渡された番号にある数字を取得します
+        /// </summary>
+        /// <param name="index">index番号</param>
+        /// <returns>指定した番号の値</returns>
+        public int? GetNumber(int index)
+        {
+            return numbers[index].Value;
+        }
+
+        /// <summary>
+        /// 引数で渡された番号にあるNumberを取得します
+        /// </summary>
+        /// <param name="i">先頭からの番号</param>
+        /// <returns>指定した番号のNumber</returns>
+        public Number this[int i] => numbers[i];
+
+
+    }
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Numbers numbers = new Numbers();
+
+            Random random = new Random();
+
+            for(int i = 0;i < 100;i++)
+            {
+
+                numbers.Append(random.Next(0,101));
+            }
+            numbers.Append(new Number(null));
+
+            numbers.SortValue();
+
+            numbers.Output();
+
+            Console.WriteLine("100番目の番号\n" + numbers.GetNumber(100));
```

Comment:
下記のように、「意図を読み取りやすいコード」を書けるようにするためにNumbersクラスにインデクサを用意したのでは？
```suggestion
            Console.WriteLine("100番目の番号\n" + numbers[100].Value);
```

Reply:
インデクサを使うという部分に重きを置いていたため、「意図を読み取りやすいコード」という部分には頭が回っていませんでした。
インデクサがある以上、GetNumber関数は必要ないと思うので関数を削除して修正しておきます。

Reply:
修正しました。